trace/id_test.go

```diff
@@ -0,0 +1,156 @@
+package trace
+
+import (
+	"log"
+	"testing"
+)
+
+const (
+	TestIDStr = "094f13e642b9b07f"
+	TestIDVal = 670776749184364671
+)
+
+func TestNewID(t *testing.T) {
+	i1 := NewID()
+	i2 := NewID()
+	if i1 == i2 {
+		t.Errorf("Two generated IDs are the same: %d, %d", i1, i2)
+	}
+	if i1 == 0 || i2 == 0 {
+		t.Errorf("One of the ids are 0: %d, %d", i1, i2)
+	}
+}
+
+func TestParseID(t *testing.T) {
+	i, err := ParseID(TestIDStr)
+	if err != nil {
+		t.Error("Error occured parsing the ID: ", err.Error())
+	}
+	if i != TestIDVal {
+		t.Errorf("The parsed value is not matching: %d", i)
+	}
+}
+
+func TestWrongParseID(t *testing.T) {
+	_, err := ParseID("x")
+	if err == nil {
+		t.Error("Error should have happened for wrong input parsing")
+	}
+}
+
+func TestString(t *testing.T) {
+	i := ID(TestIDVal)
+	if i.String() != TestIDStr {
+		t.Errorf("String ID value does not match: %s", i.String())
+	}
+}
+
+func TestEquality(t *testing.T) {
+	i1 := ID(TestIDVal)
+	i2 := ID(TestIDVal)
+	if i1 != i2 {
+		t.Errorf("IDs should be equal but they are not: %016x != %016x", uint64(i1), uint64(i2))
+	}
+	if i1.String() != i2.String() {
+		t.Errorf("IDs should be equal but they are not: %s != %s", i1.String(), i2.String())
+	}
+	i1v, err := ParseID(i1.String())
+	if err != nil {
+		t.Error("Error occured parsing the ID: ", err.Error())
+	}
+	i2v, err := ParseID(i2.String())
+	if err != nil {
+		t.Error("Error occured parsing the ID: ", err.Error())
+	}
+	if i1v != i2v {
+		t.Errorf("IDs should be equal but they are not: %s != %s", i1.String(), i2.String())
+	}
+}
+
+func TestNewRootSpanID(t *testing.T) {
+	i := NewRootSpanID()
+	if i.Trace == i.Span {
+		t.Errorf("Trace and Span should not be equal: %016x !=  %016x", uint64(i.Trace), uint64(i.Span))
+	}
+	if i.Parent != 0 {
+		t.Errorf("Root SpanID shouldn't have Parent ID: %016x != %016x", uint64(i.Parent), uint64(0))
+	}
+}
+
+func TestRootIsRoot(t *testing.T) {
+	i := NewRootSpanID()
+	log.Println(i.Empty())
```

Comment:
Please don't log in tests. If you really have to log something then use `t.Log` and `t.Logf` which would only show up if you run `go test -v`.